doc/rados/operations/crush-map-edits.rst

```diff
@@ -32,193 +33,199 @@ Pool Values`_.
 
 #. **tunables:** The preamble at the top of the map describes any *tunables*
-   that differ from the historical / legacy CRUSH behavior. These
-   correct for old bugs, optimizations, or other changes that have
-   been made over the years to improve CRUSH's behavior.
+   that are not a part of legacy CRUSH behavior. These tunables correct for old
+   bugs, optimizations, or other changes that have been made over the years to
+   improve CRUSH's behavior.
 
 #. **devices:** Devices are individual OSDs that store data.
 
-#. **types**: Bucket ``types`` define the types of buckets used in
-   your CRUSH hierarchy. Buckets consist of a hierarchical aggregation
-   of storage locations (e.g., rows, racks, chassis, hosts, etc.) and
-   their assigned weights.
+#. **types**: Bucket ``types`` define the types of buckets that are used in
+   your CRUSH hierarchy. 
 
-#. **buckets:** Once you define bucket types, you must define each node
-   in the hierarchy, its type, and which devices or other nodes it
-   contains.
+#. **buckets:** Buckets consist of a hierarchical aggregation of storage
+   locations (for example, rows, racks, chassis, hosts) and their assigned
+   weights. After you have defined bucket ``types``, you must define each node
+   in the hierarchy, its type, and which devices or other nodes it contains.
 
 #. **rules:** Rules define policy about how data is distributed across
    devices in the hierarchy.
 
-#. **choose_args:** Choose_args are alternative weights associated with
-   the hierarchy that have been adjusted to optimize data placement.  A single
-   choose_args map can be used for the entire cluster, or one can be
-   created for each individual pool.
+#. **choose_args:** ``choose_args`` are alternative weights associated with
+   the hierarchy that have been adjusted in order to optimize data placement. A
+   single ``choose_args`` map can be used for the entire cluster, or a number
+   of ``choose_args`` maps can be created such that each map is crafted for a
+   particular pool.
 
 
 .. _crushmapdevices:
 
-CRUSH Map Devices
+CRUSH-Map Devices
 -----------------
 
-Devices are individual OSDs that store data.  Usually one is defined here for each
-OSD daemon in your
-cluster.  Devices are identified by an ``id`` (a non-negative integer) and
-a ``name``, normally ``osd.N`` where ``N`` is the device id.
+Devices are individual OSDs that store data. In this section, there is usually
+one device defined for each OSD daemon in your cluster. Devices are identified
+by an ``id`` (a non-negative integer) and a ``name`` (usually ``osd.N``, where
+``N`` is the device's ``id``).
+
 
 .. _crush-map-device-class:
 
-Devices may also have a *device class* associated with them (e.g.,
-``hdd`` or ``ssd``), allowing them to be conveniently targeted by a
-crush rule.
+A device can also have a *device class* associated with it: for example,
+``hdd`` or ``ssd``. Device classes allow devices to be targeted by CRUSH rules.
```

Comment:
Add ``nvme`` just for lulz

Device classes allow CRUSH rules to select only OSDs that match desired characteristics.  One might for example want an RBD pool that lives only on SSDs and one that lives only on HDDs.

Reply:
Accepted with minor modifications.

nvme - not yet, not yet.